lib/ui/platform_dispatcher.dart

```diff
@@ -0,0 +1,1586 @@
+      devicePixelRatio: devicePixelRatio,
+      geometry: Rect.fromLTWH(left, top, width, height),
+      viewPadding: WindowPadding._(
+        top: viewPaddingTop,
+        right: viewPaddingRight,
+        bottom: viewPaddingBottom,
+        left: viewPaddingLeft,
+      ),
+      viewInsets: WindowPadding._(
+        top: viewInsetTop,
+        right: viewInsetRight,
+        bottom: viewInsetBottom,
+        left: viewInsetLeft,
+      ),
+      padding: WindowPadding._(
+        top: math.max(0.0, viewPaddingTop - viewInsetTop),
+        right: math.max(0.0, viewPaddingRight - viewInsetRight),
+        bottom: math.max(0.0, viewPaddingBottom - viewInsetBottom),
+        left: math.max(0.0, viewPaddingLeft - viewInsetLeft),
+      ),
+      systemGestureInsets: WindowPadding._(
+        top: math.max(0.0, systemGestureInsetTop),
+        right: math.max(0.0, systemGestureInsetRight),
+        bottom: math.max(0.0, systemGestureInsetBottom),
+        left: math.max(0.0, systemGestureInsetLeft),
+      ),
+    );
+    _invoke(onMetricsChanged, _onMetricsChangedZone);
+  }
+
+  /// A callback invoked when any view begins a frame.
+  ///
+  /// {@template flutter.foundation.PlatformDispatcher.onBeginFrame}
+  /// A callback that is invoked to notify the application that it is an
+  /// appropriate time to provide a scene using the [SceneBuilder] API and the
+  /// [PlatformDispatcher.render] method.
+  ///
+  /// When possible, this is driven by the hardware VSync signal of the attached
+  /// screen with the highest VSync rate. This is only called if
+  /// [PlatformDispatcher.scheduleFrame] has been called since the last time
+  /// this callback was invoked.
+  /// {@endtemplate}
+  FrameCallback? get onBeginFrame => _onBeginFrame;
+  FrameCallback? _onBeginFrame;
+  Zone _onBeginFrameZone = Zone.root;
+  set onBeginFrame(FrameCallback? callback) {
+    _onBeginFrame = callback;
+    _onBeginFrameZone = Zone.current;
+  }
+
+  // Called from the engine, via hooks.dart
+  void _beginFrame(int microseconds) {
+    _invoke1<Duration>(
+      onBeginFrame,
+      _onBeginFrameZone,
+      Duration(microseconds: microseconds),
+    );
+  }
+
+  /// {@template flutter.foundation.PlatformDispatcher.onDrawFrame}
+  /// A callback that is invoked for each frame after [onBeginFrame] has
+  /// completed and after the microtask queue has been drained.
+  ///
+  /// This can be used to implement a second phase of frame rendering that
+  /// happens after any deferred work queued by the [onBeginFrame] phase.
+  /// {@endtemplate}
+  VoidCallback? get onDrawFrame => _onDrawFrame;
+  VoidCallback? _onDrawFrame;
+  Zone _onDrawFrameZone = Zone.root;
+  set onDrawFrame(VoidCallback? callback) {
+    _onDrawFrame = callback;
+    _onDrawFrameZone = Zone.current;
+  }
+
+  // Called from the engine, via hooks.dart
+  void _drawFrame() {
+    _invoke(onDrawFrame, _onDrawFrameZone);
+  }
+
+  /// A callback that is invoked when pointer data is available.
```

Comment:
If I get a PointerData package, how do I know in which view the pointer event occurred?